### ISSUE TITLE
VMUser `targetRefs.crd[]` uses path wildcard, preventing "single-tenant view" (usage of `VMUser/vminsert` and `VMUser/vmselect`)

### ISSUE THREAD

**Issue by flokli** [VictoriaMetrics/operator#379]
I created the following `VMUser` resource:

```yaml
apiVersion: operator.victoriametrics.com/v1beta1
kind: VMUser
metadata:
  name: foo-vmalert
  namespace: default
spec:
  generatePassword: true
  targetRefs:
  - crd:
      kind: VMCluster/vminsert
      name: vmcluster
      namespace: default
    target_path_suffix: /insert/30
  - crd:
      kind: VMCluster/vmselect
      name: vmcluster
      namespace: default
    target_path_suffix: /select/30
  username: foo-vmalert
```

The idea was that this will look like a single-tenant vminsert and vmselect endpoint.

However, the rendered config looks like this:

```yaml
- url_map:
  - url_prefix: http://vminsert-vmcluster.default.svc:8480/insert/30
    src_paths:
    - /.*
  - url_prefix: http://vmselect-vmcluster.default.svc:8481/select/30
    src_paths:
    - /.*
  username: foo-vmalert
  password: XXX
```

… and vmselect queries mistakenly end up in the vminsert endpoint, and vice-versa.

Is there a way vminsert/vmselect could configure more specific `src_paths` only, those that are configured in `vminsert`/`vmselect`?

**Comment by f41gh7** [comment]
I think, in this case vmselect must serve all unmatched paths - `/.*` and vminsert only specific urls for data ingestion. It changes less frequent and doesn't have any complicated paths. 

something like: 

```yaml
apiVersion: operator.victoriametrics.com/v1beta1
kind: VMUser
metadata:
  name: foo-vmalert
  namespace: default
spec:
  generatePassword: true
  targetRefs:
  - crd:
      kind: VMCluster/vminsert
      name: vmcluster
      namespace: default
    target_path_suffix: /insert/30
    paths:
     - /prometheus/api/v1/write
     - /prometheus/api/v1/import
     - /prometheus/api/v1/import/csv
     - /prometheus/api/v1/import/prometheus
     - /prometheus/api/v1/import/native
     - /influx/.*
     - /datadog/.*
  - crd:
      kind: VMCluster/vmselect
      name: vmcluster
      namespace: default
    target_path_suffix: /select/30
  username: foo-vmalert
```

 Personally, i don't like idea to keep list of hard coded urls for vminsert/vmselect. Its hard to maintain and vmselect has complex routing paths.

**Comment by flokli** [comment]
This might still break if people plug other services into `targetRefs`.

> Personally, i don't like idea to keep list of hard coded urls for vminsert/vmselect. Its hard to maintain and vmselect has complex routing paths.

If we don't keep that list in the operator, we shift this burden down to users - asking them to maintain these paths probably also isn't nice. I'd prefer if victoria metrics operator kept a list of paths for `VMCluster/{vminsert,vmselect}`.

It shouldn't be only used to lookup the name of the service, but also keep the list of paths. It already knows a lot about VM internals, and controls the version that's deployed.

**Comment by f41gh7** [comment]
Yes, that's true. 

 So, there can be cases:
1) single crd ref, keep exist behavior. 
2) vminsert and vmselect ref, autogenerate config, that i mentioned. If pathes are empty.
3) vminsert, vmselect and other refs. Add explicit paths to the vminsert and vmselect. If pathes for vmselect and vminsert are empty.

 It looks like magic and not obvious for users, so it must be added to the doc.

**Comment by f41gh7** [comment]
Changes was included to the 0.21.0 release.

**Comment by flokli** [comment]
Thanks! And as can be seen from https://github.com/VictoriaMetrics/operator/commit/ca026b92065d1f2865ad780bc3f9a0f6fa2ecc61#diff-2b9b772fbfb5c4dcf3e619d3e5b4f9fbf3863c9388f78a7420165a5763d19c03R649-R674, path seem to be specific enough.
